src/dist/http.rs

```diff
@@ -309,83 +308,116 @@ mod server {
+        let rsa_key = rsa::RSAPrivateKey::new(&mut rng, bits)?;
+
+        let sk_pkcs8 = rsa::PrivateKeyPemEncoding::to_pem_pkcs8(&rsa_key)?;
+        let pk_pkcs8 = rsa::PublicKeyPemEncoding::to_pem_pkcs8(&*rsa_key)?;
+
+        // convert to picky
+        let sk = PrivateKey::from_pem_str(sk_pkcs8.as_str())?;
+        let pk = PublicKey::from_pem_str(pk_pkcs8.as_str())?;
+
+        let today = chrono::Utc::now().naive_utc();
+        let expires = today + chrono::Duration::days(365);
+        let start = UTCDate::new(
+            today.year() as u16,
+            today.month() as u8,
+            today.day() as u8,
+            today.time().hour() as u8,
+            today.time().minute() as u8,
+            today.time().second() as u8,
+        )
+        .unwrap();
+        let end = UTCDate::new(
+            expires.year() as u16,
+            expires.month() as u8,
+            expires.day() as u8,
+            expires.time().hour() as u8,
+            expires.time().minute() as u8,
+            expires.time().second() as u8,
+        )
+        .unwrap();
+
+        let extended_key_usage = ExtendedKeyUsage::new(vec![picky::oids::kp_server_auth()]);
+
+        let name = addr.to_string();
+
+        let issuer_name = DirectoryName::new_common_name(name.clone());
+        let subject_name = DirectoryName::new_common_name(name);
+        let octets = match addr.ip() {
+            IpAddr::V4(inner) => inner.octets().to_vec(),
+            IpAddr::V6(inner) => inner.octets().to_vec(),
+        };
+        let subject_alt_name = GeneralNames::new(picky::x509::name::GeneralName::IpAddress(octets));
+
+        let cert = CertificateBuilder::new()
+            .validity(start, end)
+            .key_usage(KeyUsage::new(1))
+            .subject(subject_name, pk)
+            .subject_alt_name(subject_alt_name.clone())
+            .serial_number(vec![0])
+            .signature_hash_type(SignatureAlgorithm::RsaPkcs1v15(HashAlgorithm::SHA1))
+            .key_id_gen_method(KeyIdGenMethod::SPKValueHashedLeftmost160(
+                HashAlgorithm::SHA2_256,
+            ))
+            .extended_key_usage(extended_key_usage.clone())
+            .self_signed(issuer_name, &sk)
+            .build()?;
+
+        // TODO exists to assure compat with the previously created cert
+        // TODO but imho this can be removed eventually
+        let cert = {
+            use picky_asn1_x509::certificate::Certificate;
+
+            let mut certificate = Certificate::from(cert);
+            let inner = &mut certificate.tbs_certificate;
+            let extensions = inner.extensions.deref_mut();
+
+            // let basic = dbg!(picky::x509::Extension::new_key_usage(KeyUsage::new(0)));
+            let subject_alt_name = picky::x509::Extension::new_subject_alt_name(subject_alt_name);
+            let extended_key_usage =
+                picky::x509::Extension::new_extended_key_usage(extended_key_usage);
+
+            *extensions = Extensions(vec![
```

Comment:
This is probably incompatible with how picky works wrt self signatures:

Signing logic: https://github.com/Devolutions/picky-rs/blob/2ec6f13223100d432681f16892b11580174a4293/picky/src/x509/certificate.rs#L915-L937

It calls `signature_hash_type.sign(&tbs_der, issuer_key)` but the `tbs_der` is a `picky_asn1_der::to_vec`-serialized certificate which includes previously collected `extensions` during certificate build.

The prior extensions are serialized as part of the certificate and that payload is signed but here we just replace it with another set of extensions, thus invalidating the signature (we alter data we just signed).